### DISCUSSION TITLE
IPad

### DISCUSSION THREAD

**Discussion by PaulLBos** [pybricks/support#657] [Q&A]
Can I run pybricks on an IPad?
Safari does not support wed Bluetooth 
Question Is there a web browser that will run on an IPad that does support Web Bluetooth?

**Comment by dlech** [comment]
There are some apps that in theory support web Bluetooth on iOS. We have not tested any of them, so we can't say if it will work or not.

**Reply by PaulLBos** [reply]
Please give me a name of a posable app and I will try it and let you know.

**Reply by laurensvalk** [reply]
There’s an app called WebBLE that used to work but not anymore. It might be something we could fix, but our focus has been the desktop so far.

![97F7727C-624D-470E-9798-48D448B04AF4](https://user-images.githubusercontent.com/12326241/170860201-ff3b47be-b7ba-45ba-951b-af97fcfa4af2.png)

**Comment by dlech** [comment]
Bluefy is another app that claims to support Web Bluetooth.

**Comment by laurensvalk** [comment]
Guess what - the author of WebBLE has an open issue to support Pybricks, although it hasn't been updated in a while. https://github.com/daphtdazz/WebBLE/issues/42

If we find that we have a lot of iPad users, maybe we can help them. Since it used to work, maybe it's a matter of working around that availability check error in the screenshot above.

Now that we switched to the Monaco editor, maybe using an iPad isn't even so bad. The explicit toggle keyboard button is certainly helpful.

And, for bonus points... release a version that points to `code.pybricks.com` instead of acting as a generic browser. I'm not sure we can maintain this though.

**Reply by daphtdazz** [reply]
WebBLE now supports PyBricks, I pushed v1.7.0 this week to the App store which adds support.

**Reply by laurensvalk** [reply]
Very nice, and thanks for letting us know!

I love the idea of hosting a version in the app store as a way of supporting the project. Maybe we could do that too, with a fair share going back to `WebBLE` as well.

**Reply by laurensvalk** [reply]
Which LEGO hubs have you tried, @daphtdazz ? Thanks!

**Reply by daphtdazz** [reply]
Just the Technic hub (the only one I had!).

**Reply by daphtdazz** [reply]
The codebase for WebBLE is set up so that it should be quite easy to create your own version that just renders `code.pybricks.com` and doesn't have the navigation tools etc (or maybe limited ones) and that you could brand your own way (it's all open source!). However as you point out you'd then have to maintain it and you'd probably want to keep in sync with any changes that I push since I do occasionally fix bugs, so I'm not sure it would be really worth it for the small amount of revenue you might get for it (if you were thinking of charging for it). So I'd probably just say let's work together to make sure WebBLE gives a good enough experience (most of my proceeds which is about £300 a month from WebBLE come from people using it for vaping apps, which is a bit galling), and I'll try to be a bit more responsive than I have been!

**Reply by laurensvalk** [reply]
> Just the Technic hub (the only one I had!).

Good to know. That sounds like it means the BLE firmware updater is working as well.

SPIKE hubs use USB with a web-DFU driver, which is probably (?) out of scope.

> The codebase for WebBLE is set up (...)

This sounds good. I don't really know what we'll do yet -- in the meanwhile we'll send people your way when they ask :)

There's a pretty big update coming for Pybricks. It's a bit under wraps, but let's just say it doesn't require typing code... So this makes the iPad a neat option. Good timing! :smile: 

> (...) doesn't have the navigation tools etc (or maybe limited ones) (...)

Maybe the Pybricks `app` sub menu could have a `try the beta version`, linking to https://beta.pybricks.com/ and vice versa, which might be a good idea anyway.

**Comment by hyper3dp** [comment]
I got it working currently with blueify and pybricks beta 
<img width="1180" height="820" alt="IMG_1848" src="https://github.com/user-attachments/assets/e8118f83-8eba-412f-b4ad-bee4c5bb6493" />

**Reply by hyper3dp** [reply]
As early may 2026 this still works

**Reply by hyper3dp** [reply]
Update I made a yt vid explaining how to do it https://youtu.be/APo-4Kb40Uw

**Reply by laurensvalk** [reply]
Nicely done!

**Comment by bcfletcher** [comment]
i can connect to the Spike Hub (see pix) but there issues with load python files (see pix) using Bluefy on my iPad. it didn't understand the name of the hub but it did find the address. however, it couldn't input saved files due to a mime type error.

i tried renaming the file to a .txt extension (see ue_menu_c_rev54.txt) but pybricks will not let me select that file type to open. :(


notice on the 2nd image it shows the hub connected but not the name or the Bluetooh MAC of the device. but when discovering it showed its BlueTooth MAC but not name


i tested with both the new production pybricks, beta, and labs.

prod & beta same results
labs did show the hub name along with the Bluetooh Mac (see last image)

but prod, beta, and labs all have the  myfile.py unsupported file type text/x-python-script MIME Type error



<img width="820" height="1180" alt="Screenshot 2026-06-10 at 21 54 38" src="https://github.com/user-attachments/assets/de3bc9e0-9117-4ecf-9995-4dffa71f26a9" />


<img width="820" height="1180" alt="Screenshot 2026-06-10 at 22 11 25" src="https://github.com/user-attachments/assets/b3fd8f4d-373e-4cbe-a2dc-141084daa056" />


<img width="820" height="1180" alt="IMG_1703" src="https://github.com/user-attachments/assets/f0c33d67-e66e-46cf-92fd-df1b38db1ca2" />



<img width="820" height="1180" alt="Screenshot 2026-06-10 at 22 08 22" src="https://github.com/user-attachments/assets/25156334-e936-4bf6-9b0b-8c0f3df19278" />


<img width="820" height="1180" alt="Screenshot 2026-06-10 at 22 18 16" src="https://github.com/user-attachments/assets/103a6641-8258-4fe5-9bbc-6cde3a5a8885" />

**Reply by hyper3dp** [reply]
Your not on pybricks beta

**Reply by hyper3dp** [reply]
The first time you connect to the hub its name is a long string

**Reply by hyper3dp** [reply]
Also have you installed the pybricks os on your computer

**Reply by hyper3dp** [reply]
You need to install that

**Reply by bcfletcher** [reply]
i tested pybricks prod, beta, and labs versions. yes i have it installed on my computers (mac and PC) but having pybricks installed on my computer isn't relevant to using pybricks on my iPad (other than perhaps to update firmware on the hub).

**Reply by hyper3dp** [reply]
Just try using blocks

**Reply by bcfletcher** [reply]
pybricks block coding stores the files as python files since they are actually python with a large JSON object as a comment in the first line for block rendering.

while pybricks block coding is awesome and FAR superior to the crap Lego wrote - we don't use it. i have middle schoolers and  they are using python exclusively and very successfully for both our FLL teams.

**Reply by laurensvalk** [reply]
Thanks @bcfletcher for reporting - Yes, I agree Python is just as important too.

I will add your case to https://github.com/pybricks/support/issues/1436, which we should be able to fix quite soon.

The Bluetooth name not showing is probably out of our control on that browser. But since we do filter for hubs that run Pybricks, you should just be able to pick the one you see.

**Reply by bcfletcher** [reply]
thanks.
